geopandas/geodataframe.py

```diff
@@ -631,6 +631,31 @@ def _dataframe_set_geometry(self, col, drop=False, inplace=False, crs=None):
                          " DataFrame to GeoDataFrame")
     return GeoDataFrame(self, geometry=col, crs=crs)
 
+
+def concat(L, axis=0):
+    """ Concatenate multiple GeoDataFrames or GeoSeries together
+
+    Currently only works for axis=0
+    """
+    if axis != 0:
+        raise NotImplementedError("Only axis=0 supported")
+
+    if isinstance(L[0], GeoDataFrame):
+        name = L[0]._geometry_column_name
+    else:
+        name = L[0].name
+
+    geometry = vectorized.concat(df._geometry_array for df in L)
+    if isinstance(L[0], GeoDataFrame):
+        L = [df.drop(name, axis=1) for df in L]
+        new = pd.concat(L)
```

Comment:
maybe we can pass through the `ignore_index` arg ?

Reply:
Now passing through ignore_index, but not yet testing this

Reply:
Now testing, although in the future we should probably find some way to test against proper pandas behavior with a better test suite.